src/typechecker/lisp-type.lisp

```diff
@@ -49,6 +93,41 @@ USE-FUNCTION-ENTRIES specifies whether to emit FUNCTION-ENTRY for functions, emi
       ((function-type-p ty)
        'function-entry)
 
+      ((typep (tapp-from ty) 'tycon)
+       (let ((from (tycon-name (tapp-from ty)))
+             (to   (tapp-to ty)))
+         (cond
+           ;; First we deal with specialized parametric types.
+           ;;
+           ;; (Complex :t)
+           ((complex-tycon-p from)
+            (let ((lisp-to (lisp-type to env)))
+              (cond
+                ((member lisp-to
+                         *specialized-complex-part-types-considered*
+                         :test #'lisp-type=)
+                 `(cl:complex ,lisp-to))
```

Comment:
the `cl:` is unnecessary

Reply:
i put it there since there's a coalton/whatever:complex, and this file deals with both lisp types and coalton tycons